docs/reference-metrics-opentelemetry.md

```diff
@@ -8,6 +8,153 @@ Pulsar exposes the following OpenTelemetry metrics.
 
 ## Broker
 
+### Topic Messaging metrics
```

Comment:
Do we need to consider Sync vs Async instruments for this `Type`?

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#synchronous-and-asynchronous-instruments

Reply:
The implementation https://github.com/apache/pulsar/pull/22467 uses asynchronous instruments, due to performance and resource efficiency reasons (more details in that PR). This is a detail, however, so it doesn't need to be in the metrics reference documentation.

Furthermore, from the users point of view, this is transparent. Instrument types do not impact the way the user needs to collect or interpret these values.